### PR TITLE
Use grid for layout, fixing scroll and resize on select

### PR DESCRIPTION
**Pull request by garrett**
This also paves the path for better mobile support too.

### CODE REVIEW

**Comment by garrett** [comment]
This is a draft PR as a preview; I'm still working on it, but wanted to share what I have so far.

I'm planning on trying out subgrid instead and also intend to clean it up a bit more.

**Comment by garrett** [comment]
I marked it for review yesterday. This is just a comment to say I'm waiting on review now, from @jelly and @allisonkarlitskaya, as I guess GitHub doesn't make that clear. (I also mentioned it in Matrix yesterday evening though. :wink:)

**Comment by jelly** [comment]
Some interesting obversations:

Previously we had some grey area around the white folder view

image


This is now gone:

image

That seems unexpected to me?

**Comment by jelly** [comment]
Getting mobile further to work is a follow up I suppose:

image

I wonder if the details should be shown on top (The sidebar).

**Review by jelly — COMMENTED** [review]
I've made some comments in the PR

**Comment by garrett** [comment]
> Getting mobile further to work is a follow up I suppose

Correct. I mentioned that in Matrix yesterday and today, both times when linking this. We also made the decision a while ago to work toward mobile, but not make sure it works yet. We're planning on doing a mobile push as a follow-up in the near future.

We'll need to rework the way the data is shown (as-is will absolutely not work in mobile), and that requires more design and implementation. It shouldn't block the layout fixes. What's in this PR improves over what is in main.

**Comment by garrett** [comment]
> I wonder if the details should be shown on top (The sidebar).

No.

1. The details _have to_ come after what you're selecting, not before. (Very basic design rule: where what's to the left/top affects things that come after it, not the other way around. RTL flips left/right.)
2. If it's above and changes height, it shifts the location of everything after.

What I'm suggesting is a statusbar area that has basic details only and could possibly be expanded to show more. But that needs more design and implementation work. For now, I have it limited to a size and have it scrollable as a workaround for the time being. (It's already better than the status quo for mobile, as you can actually see and interact with it versus a huge page of scrolling back and forth.)

**Comment by garrett** [comment]
> I've made some comments in the PR

Really? Where. I don't see them. (You're talking about inline comments in the code diff view, right?)

**Review by jelly — APPROVED** [review]

File: src/files-card-body.jsx [hunk 1]
```diff
@@ -363,7 +363,7 @@ export const FilesCardBody = ({
     });
 
     return (
-        <div id={files_parent_id} ref={folderViewRef}>
+        <div id={files_parent_id} className="fileview-wrapper" ref={folderViewRef}>
```
Comment (jelly) [inline src/files-card-body.jsx:366]:
```
# /work/make-checkout-workdir/src/files-card-body.jsx
#   366:64  warning  Prop `ref` must be placed on a new line  react/jsx-max-props-per-line
```

Other then that its good.

**Review by jelly — CHANGES_REQUESTED** [review]

File: src/files-card-body.jsx (hunk 1 above)
Comment (garrett) [reply src/files-card-body.jsx]: So is this "correct"? f3404e00fa61c1bde789ec4f17627547f51e0532

(We'll squash all this down when it's time for merging.)

**Comment by garrett** [comment]
19cc7db5a2e194468067883746ebbca81963418d caused a conflict, so I undid everything from that commit except for the test, as this PR doesn't have the issue that the commit was fixing.

(I did another rebase and fixed the conflicts, in other words. :wink: The difference is that we gained the test.)

**Comment by jelly** [comment]
> [19cc7db](https://github.com/cockpit-project/cockpit-files/commit/19cc7db5a2e194468067883746ebbca81963418d) caused a conflict, so I undid everything from that commit except for the test, as this PR doesn't have the issue that the commit was fixing.
> 
> (I did another rebase and fixed the conflicts, in other words. 😉 The difference is that we gained the test.)

Thanks!

**Review by jelly — APPROVED** [review]

**Pull request merged**